geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/PyramidFactory.scala

```diff
@@ -334,10 +334,31 @@ class PyramidFactory(collectionId: String, datasetId: String, catalogApi: Catalo
 
             tracker.addInputProducts(collectionId, features.keys.toList.asJava)
 
-            val featureIntersections = for {
-              feature <- sc.parallelize(features.values.toSeq, math.max(1, features.size / 10))
-              reprojectedFeature = feature.reproject(LatLng, boundingBox.crs)
-            } yield Feature(reprojectedFeature intersection multiPolygon, reprojectedFeature.data.toLocalDate.atStartOfDay(UTC))
+            val featuresRDD = sc.parallelize(features.values.toSeq, math.max(1, features.size / 10))
+            val featureIntersections = featuresRDD.flatMap(feature => {
+              val reprojectedFeature = feature.reproject(LatLng, boundingBox.crs)
+              // In test over England, there where up to 0.003 deviations on long line segments due to curvature change between CRS
+              val multiPolygonBuffered = multiPolygon
+                .reproject(polygons_crs, LatLng)
+                .buffer(0.006)
+                .reproject(LatLng, polygons_crs)
+              val intersect = reprojectedFeature intersection multiPolygonBuffered
+
+              if (intersect.isEmpty) {
+                logger.debug(s"shub returned a Feature that actually does not intersect with our requested polygons.")
+                Seq()
+              } else
+                Seq(Feature(intersect, reprojectedFeature.data.toLocalDate.atStartOfDay(UTC)))
```

Comment:
You return a collection with 0 or 1 element(s): that's an `Option`. You can return `None` and `Some(Feature(...))` instead.

Reply:
Did not know, changed!